app/waf/src/controllers/declaration.controller.ts

```diff
@@ -333,4 +342,56 @@ export class DeclarationController extends BaseController {
       .declarations(applicationId)
       .delete({and: [{id: declarationId}, {tenantId: await this.tenantId}]});
   }
+
+  @post(
+    prefix +
+      '/applications/{applicationId}/declarations/{declarationId}/deploy',
```

Comment:
As I recommended before, I don't think to prefix `/applications/{applicationId}` to the API is a good idea. 

Indeed, the declaration has a relation with the application... In fact, the declaration has a relationship with all elements of the TREE... 

The relation link has already been saved in declaration model, we/user can get it through the query string(`GET /declarations?applicationId=xxx` or `GET /applications/{applicationId}/delcarations`). 
That's not meaning we need to bind the relationship everywhere, even into the API call. 

Think that: when the application TREE's configuration is newer than a declaration generated before, to prefix the applicationId to that declaration's deployment would be confused, no matter succeed or fail. 

I think declaration is a standalone existence, which can exist or be operated on with no regarding the application which it is generated from...